src/esp/io/JsonEspTypes.h

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+/** @file
+ * @brief See JsonAllTypes.h. Don't include this header directly in user code.
+ */
+
+#include "JsonBuiltinTypes.h"
+#include "JsonMagnumTypes.h"
+
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+#include "esp/core/esp.h"
+#include "esp/gfx/replay/Keyframe.h"
+
+namespace esp {
+namespace io {
+
+inline RJsonValue ToRJsonValue(const esp::vec3f& vec,
```

Comment:
All the ToRJsonValue/FromJsonValue here are inline. This has nothing to do with runtime perf. My thinking is: (1) it's tedious to do declarations *and* definitions for all these small functions (and all the other types we'll add here), so let's just do inline definitions, and (2) these functions are all really small so it's not expensive to inline them.

I have at least one more complex object coming soon (later PR) that I won't inline because it feels like too much code to inline.